src/state/zustand/swapperStore/amountSelectors.ts

```diff
@@ -0,0 +1,357 @@
+import type { Selector } from 'reselect'
+import { createSelector } from 'reselect'
+import { TradeAmountInputField } from 'components/Trade/types'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit, toBaseUnit } from 'lib/math'
+import {
+  selectAction,
+  selectAmount,
+  selectBuyAsset,
+  selectBuyAssetFiatRate,
+  selectSelectedCurrencyToUsdRate,
+  selectSellAsset,
+  selectSellAssetFiatRate,
+} from 'state/zustand/swapperStore/selectors'
+import type { SwapperState } from 'state/zustand/swapperStore/types'
+
+const selectAssetPriceRatio = createSelector(
+  selectBuyAssetFiatRate,
+  selectSellAssetFiatRate,
+  (buyAssetFiatRate, sellAssetFiatRate) => {
+    return bnOrZero(buyAssetFiatRate).dividedBy(bnOrZero(sellAssetFiatRate)).toFixed()
```

Comment:
Can ditch the inner `bnOrZero()` here since we're in the context of a BN method

Reply:
https://github.com/shapeshift/web/pull/4138/commits/4055a9f837db9ae2527b0b3e36fc8fb4aacb8cc9